src/components/Page.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from 'react';
+import { Route, RouteProps } from 'react-router-dom';
+
+interface PageProps extends RouteProps {
+  title: string;
+  setHeader: React.Dispatch<React.SetStateAction<string>>;
+}
+
+const Page: React.FC<PageProps> = (props: PageProps) => {

```

Comment:
I'm not sure what the best name for this class would be, but I feel like it's not Page. In a web app I'd expect a Page class to model a web page, but this feels like an extension for the Route component that adds a title. `TitledRoute` maybe? `NamedRoute`?

Reply:
Ok I'll rename it 'TitledRoute'